### PR TITLE
Test error response when zendesk signature invalid

### PR DESCRIPTION
**Pull request by SamBerry-GDS**
Adds test an error is returned for invalid signature calls for TT2-16

### CODE REVIEW

File: integration-tests/zendeskRequestAPI.spec.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import axios from 'axios'
+
+describe('Zendesk request API', () => {
+  it('returns a Invalid request source on invalid Zendesk Webhook Signature', async () => {
```
Comment (rikumolugds) [inline integration-tests/zendeskRequestAPI.spec.ts:4]: `API Gateway returns an Invalid request source when request has invalid Zendesk Webhook Signature`?

File: integration-tests/zendeskRequestAPI.spec.ts [hunk 2]
```diff
@@ -0,0 +1,50 @@
+import axios from 'axios'
+
+describe('Zendesk request API', () => {
```
Comment (rikumolugds) [inline integration-tests/zendeskRequestAPI.spec.ts:3]: `Zendesk request integrity` perhaps?

File: integration-tests/zendeskRequestAPI.spec.ts [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+
+    const invalidRequestError = async () => {
+      return axios({
+        url: `https://${awsBaseUrl}/Stage/zendesk-webhook`,
+        method: 'POST',
+        headers,
+        data
+      })
+        .then(() => {
+          throw 'Axios did not recieve an error'
```
Comment (rikumolugds) [inline integration-tests/zendeskRequestAPI.spec.ts:36]: `Gateway did not return an error` just to make it client agnostic?

File: integration-tests/zendeskRequestAPI.spec.ts [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        })
+        .catch((error) => {
+          if (error.response) {
+            return error.response
+          }
+          throw 'Axios did not recieve a response error'
```
Comment (rikumolugds) [inline integration-tests/zendeskRequestAPI.spec.ts:42]: Same as above.

File: integration-tests/zendeskRequestAPI.spec.ts [hunk 5]
```diff
@@ -0,0 +1,50 @@
...
+        })
+    }
+
+    const errorResponse = await invalidRequestError()
+    expect(errorResponse.status).toBe(400)
```
Comment (rikumolugds) [inline integration-tests/zendeskRequestAPI.spec.ts:47]: The acceptance criteria states unauthorised, suggesting a 401. Do we need to update the a/c? Also, there's a final a/c about the attempt being logged, not sure if that should be in Cloudwatch?

Comment (SamBerry-GDS) [reply integration-tests/zendeskRequestAPI.spec.ts:47]: We could - a 401 suggests you should re-authenticate which isn't how this works though.  I think a 400 is probably right for whats actually happening in this case.

**Review by rikumolugds — APPROVED** [review]

**Pull request merged**